resin/models/device.py

```diff
@@ -686,34 +688,37 @@ def disable_device_url(self, uuid):
             endpoint=self.settings.get('pine_endpoint')
         )
 
-    def set_to_build(self, uuid, build):
+    def set_to_build(self, uuid, build_commit_hash):
         """
-        Set a device to specific build id.
+        Set a device to specific build commit hash.
 
         Args:
             uuid (str): device uuid.
-            build (str): build id.
+            build_commit_hash (str): build commit hash.
 
         Raises:
             DeviceNotFound: if device couldn't be found.
             ApplicationNotFound: if application couldn't be found.
             IncompatibleApplication: if moving a device to an application with different device-type.
 
         Examples:
-            >> > resin.models.device.set_to_build('8deb12a58e3b6d3920db1c2b6303d1ff32f23d5ab99781ce1dde6876e8d143', '123098')
+            >> > resin.models.device.set_to_build('4457d5db93be458270666ef8b8157c66', '950eac2d3efce555490c96e7c9b55c37b385acb6')
             'OK'
 
         """
 
         if not self.has(uuid):
             raise exceptions.DeviceNotFound(uuid)
 
+        build_id = self.build.get_by_commit(build_commit_hash)[0]['id']
```

Comment:
The API doesn't guarantee only one build per commit hash (as far as I can tell), it only guarantees one _successful_ build per commit hash: https://github.com/resin-io/resin-api/blob/master/src/models/resin.sbvr#L691.

I suspect if you have a failed and then a successful build for the same commit hash, this will get the build id for the failed one, which won't work. You need to filter for success here.

Reply:
@pimterry makes sense, I've added a check to make sure users can only set to a valid build.